File: PY/Bryann Valderrama/Geometria/perimeterUsingBlocks.py
'''
Hallar el minimo perimetro de una figura usando bloques
'''

import math
from collections import namedtuple
from sys import stdin, stdout
rl = stdin.readline
wr = stdout.write


def minPerimeter(n):
    l = int(math.sqrt(n))
    sq = l * l
    if sq == n:
        return l * 4
    else:
        row = n // l
        perimeter = 2 * (l + row)
        if n % l != 0:
            perimeter += 2
        return perimeter


n = int(rl())
wr(f'{minPerimeter(n)}')

File: Codigos estudiantes por lenguaje/Java/Daniel Cangrejo/Geometria/Perimeter_Using_Blocks.java

public class Main {

	public static void main(String[] args) {
		int n = 10;
		System.out.println(minPerimeter(n));

	}
	static long minPerimeter(int n){
		int res=(int)Math.sqrt(n);
		int sq = res * res;
		if(sq==n) {
			return res*4;
		}else {
			long row = n/res;
			long perimeter = 2*(res+row);
			if(n%res!=0){
				perimeter += 2;
			}
			return perimeter;
		}
	}

}
